### PR TITLE
feat: added certificate manager support

### LINKED ISSUE #267: certificate_map support
### TL;DR

If supported already, I wonder how to use `certificate_map` with this module (serverless_negs)? Specifically [this](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_target_https_proxy#certificate_map). Essentially the last step in [this guide](https://cloud.google.com/certificate-manager/docs/deploy-google-managed-dns-auth#attach_the_certificate_map_to_the_target_proxy) is what I'm attempting to get done with the module.

Or is this supposed to be done through the resource `google_certificate_manager_certificate_map` using the `gclb_targets` block and `target_https_proxy` according to this [link](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/certificate_manager_certificate_map#nested_gclb_targets).

### Terraform Resources

```markdown
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_target_https_proxy#certificate_map
```

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by imrannayer**
fix #267

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@imrannayer
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: autogen/main.tf.tmpl [hunk 1]
```diff
@@ -25,6 +25,8 @@ locals {
   {% if not serverless %}{# Serverless NEGs don't support health checks #}
   health_checked_backends = { for backend_index, backend_value in var.backends : backend_index => backend_value if backend_value["health_check"] != null }  
   {% endif %}
+
+  certificate_map = var.certificate_map != null ? "//certificatemanager.googleapis.com/${var.certificate_map}" : null
```
Comment (g-awmalik) [inline autogen/main.tf.tmpl:29]: If this local isn't being used in multiple places, can we just set this directly on the `certificate_map` argument?

**Review by g-awmalik — COMMENTED** [review]
LGTM, just one nit if you can respond to that.

File: autogen/main.tf.tmpl (hunk 1 above)
Comment (imrannayer) [reply autogen/main.tf.tmpl:29]: Sure

**Comment by comment-bot-dev** [comment]
@imrannayer
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**